launcher/LaunchController.cpp

```diff
@@ -249,17 +257,30 @@ void LaunchController::login()
                 progDialog.execWithTask(task.get());
                 continue;
             }
-            // FIXME: this is missing - the meaning is that the account is queued for refresh and we should wait for that
-            /*
-            case AccountState::Queued: {
-                return;
-            }
-            */
             case AccountState::Expired: {
-                auto errorString = tr("The account has expired and needs to be logged into manually again.");
-                QMessageBox::warning(m_parentWidget, tr("Account refresh failed"), errorString, QMessageBox::StandardButton::Ok,
-                                     QMessageBox::StandardButton::Ok);
-                emitFailed(errorString);
+                auto errorString = tr("The account has expired and needs to be logged into manually. Press OK to log in again.");
+                auto button = QMessageBox::warning(m_parentWidget, tr("Account refresh failed"), errorString,
+                                                   QMessageBox::StandardButton::Ok | QMessageBox::StandardButton::Cancel,
+                                                   QMessageBox::StandardButton::Ok);
```

Comment:
A Yes/No messagebox might be better UX.

Question could be `The account has expired and needs to be reauthenticated. Do you want to reauthenticate this account?`